x-pack/plugin/eql/qa/common/src/main/java/org/elasticsearch/test/eql/CommonEqlActionTestCase.java

```diff
@@ -39,13 +46,20 @@ public static void checkForSnapshot() {
         assumeTrue("Only works on snapshot builds for now", Build.CURRENT.isSnapshot());
     }
 
-    @Before
+    private static boolean isSetUp = false;
+    private static int counter = 0;
+
     @SuppressWarnings("unchecked")
-    public void setUpData() throws Exception {
-        // Insert test data
-        ObjectMapper mapper = new ObjectMapper();
-        BulkRequestBuilder bulkBuilder = client().prepareBulk();
-        try (XContentParser parser = createParser(JsonXContent.jsonXContent, EqlActionIT.class.getResourceAsStream("/test_data.json"))) {
+    private static synchronized void setupData(CommonEqlActionTestCase tc) throws Exception {
```

Comment:
The client in the ESRestTestCase base class is created on @Before, so this is done in order to get the initialized client but still synchronize the setup and do it once in case the tests are run in parallel.